pkg/git/git.go

```diff
@@ -0,0 +1,61 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+package git
+
+import (
+	"fmt"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"regexp"
+
+	gitpackage "github.com/go-git/go-git/v5"
+	//"github.com/go-git/go-git/v5/plumbing/transport/http"
+)
+
+const defaultRemote = "origin"
+var (
+	// UrlRegexFormat matches the url of SSH git repository
+	httpUrlRegexFormat = regexp.MustCompile(`^((http:\/\/)|(https:\/\/))?([a-zA-Z0-9]([a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]{2,6}(\/)`)
+	sshUrlRegexFormat = regexp.MustCompile(`(ssh:\/\/[\w\d\.]+|.+@?.+\..+:)(:[\d]+){0,1}\/*(.*)`)
+	httpsPublicUrlRegexFormat = regexp.MustCompile(`(https:\/\/)?(github\.)+com(\/)`)
+	sshPublicUrlRegexFormat = regexp.MustCompile(`(git+@github.com:)(:[\d]+){0,1}/*(.*)`)
+)
+
+func ListRemote(urlPath string) error {
+	repoKind := ValidateGitURLFormat(urlPath)
+	if repoKind == "public" {
+		repo := gitpackage.NewRemote(memory.NewStorage(), &config.RemoteConfig{
+			Name: defaultRemote,
+			URLs: []string{urlPath},
+		})
+		// Note: When the urlPath is an valid public path, however, this path doesn't exist, func will return `authentication required`, this maybe mislead
+		// So convert this error message to `repository not found`
+		_, err := repo.List(&gitpackage.ListOptions{})
+		if err != nil && err == transport.ErrAuthenticationRequired {
+			return fmt.Errorf("repository not found")
+		} else if err !=nil {
```

Comment:
Maybe we can double check the `gofmt` settings of your machine.
```suggestion
		} else if err != nil {
```

Reply:
@HeavyWombat Thanks for your suggestions! Now, I have checked all go filed my added used `gofmt` tool.